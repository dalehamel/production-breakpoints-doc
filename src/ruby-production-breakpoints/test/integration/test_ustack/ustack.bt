usdt::ustack_rb:ustack_ustack_test
{
  printf("%s%s%s%s%s%s\n", str(arg0), str(arg1),
                            str(arg2), str(arg3), str(arg4), str(arg5));
  exit();
}
