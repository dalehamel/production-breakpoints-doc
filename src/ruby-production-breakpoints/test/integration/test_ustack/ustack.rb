# frozen_string_literal: true

require 'ruby-production-breakpoints'
STDOUT.sync = true

class UstackTestClass
  def poke
    a = 1
    puts 'ouch'
  end
end

ProductionBreakpoints.config_path = 'ustack_test.json'
ProductionBreakpoints.sync!

ustack_test = UstackTestClass.new

Signal.trap('USR2') do
  ustack_test.poke
end

loop do
  tp = ProductionBreakpoints.installed_breakpoints[:ustack_test].tracepoint
  puts tp.enabled?
  sleep 1
end
