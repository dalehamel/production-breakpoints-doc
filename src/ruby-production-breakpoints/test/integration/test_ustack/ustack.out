Attaching 1 probe...
{"ustack.rb:22":"`<main>'","base.rb:37":"`block (2 levels) in istall'","ustack.rb:9":"`poke'","ustack.rb:19":"`block in <main>","ustack.rb:25":"`block in <main>'"}
